typed-racket-lib/typed-racket/logic/proves.rkt

```diff
@@ -0,0 +1,20 @@
+#lang racket/base
+
+(require racket/lazy-require)
+
+(lazy-require
+ ("../typecheck/tc-envops.rkt" (implies-in-env?))
+ ("../typecheck/tc-subst.rkt" (instantiate-rep/obj)))
+
+(provide (rename-out [implies-in-env?-for-stupid-infer-unit
+                      implies-in-env?]
+                     [instantiate-rep/obj-for-stupid-infer-unit
+                      instantiate-rep/obj]))
+
+;; the units for infer currently make it impossible to do
+;; lazy requires
+(define (implies-in-env?-for-stupid-infer-unit env p q)
```

Comment:
They're lazy required to avoid cycles -- they can't appear at the top level w/o evaluating the lazy reference, so I eta expanded them.